internal/wrappers/bfl-http.go

```diff
@@ -0,0 +1,71 @@
+package wrappers
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	commonParams "github.com/checkmarx/ast-cli/internal/params"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+)
+
+const (
+	failedToParseBflResponse = "Failed to parse BFL response."
+)
+
+type BflHTTPWrapper struct {
+	path string
+}
+
+func NewBflHTTPWrapper(path string) BflWrapper {
+	return &BflHTTPWrapper{
+		path: path,
+	}
+}
+
+func (r *BflHTTPWrapper) GetBflByScanIDAndQueryID(params map[string]string) (
+	*BFLResponseModel,
+	*WebError,
+	error,
+) {
+	clientTimeout := viper.GetUint(commonParams.ClientTimeoutKey)
+	log.Println(fmt.Sprintf("Fetching the best fix location for QueryID: %s", params[commonParams.QueryIDQueryParam]))
+	resp, err := SendHTTPRequestWithQueryParams(http.MethodGet, r.path, params, nil, clientTimeout)
+	return handleBflResponseWithBody(resp, err)
```

Comment:
Should we pass the error to another method instead of handling it immediately?

Reply:
I didn't understand this. can you explain again? This was again done as per the already existing way.

Reply:
I meant I don't think this

```
resp, err := SendHTTPRequestWithQueryParams(http.MethodGet, r.path, params, nil, clientTimeout)
return handleBflResponseWithBody(resp, err)
```

is correct as we are passing the err inside the func to handle it there, we can just do

```
resp, err := SendHTTPRequestWithQueryParams(http.MethodGet, r.path, params, nil, clientTimeout)
if err != nil {
    ...
}
return handleBflResponseWithBody(resp)
```

However, I see that this pattern is used in other calls so let's not block the PR for this.